### ISSUE TITLE
Plan to support elasticsearch v7.4.0

### ISSUE THREAD

**Issue by Vimos** [wikimedia/search-extra#4]
I am trying to load `enwiki-20191007-cirrussearch-content.json.gz` into `elasticsearch 7.4.0`. But I find the plugin `org.wikimedia.search:extra` does not have a version supporting 7.4.0.

When I try to compile from source, I find the API changed a lot and may take some time to adapt.
Is there a plan to support `7.4.0`?

**Comment by Vimos** [comment]
A working version for now is in this blog: [https://www.elastic.co/blog/loading-wikipedia]

**Comment by nomoa** [comment]
Yes we will support elasticsearch 7 at some point but we don't have yet a clear timeline in mind.

**Comment by Vimos** [comment]
Great! I am really lost in the source code for now.

**Comment by nomoa** [comment]
Jumping directly from 6.5.4 to 7.4.0 might be a bit overwhelming indeed, I found that making small increments (6.5->6.6->6.7->7.0->..) often led to more comprehensive patches.

**Comment by Vimos** [comment]
Thanks! I'll try to read some code. Package `org.wikimedia.search.extra.latency;` has many code to migrate due to the removal of `org.elasticsearch.action.Action`.
